client/src/dataset/components/HDF5ParamsForm.tsx

```diff
@@ -90,13 +98,27 @@ const HDF5ParamsForm: React.FC<MergedProps> = ({
                 {dsPathInput}
             </Form.Field>
             {warning}
+            <Reshape navShape={values.nav_shape} sigShape={values.sig_shape} syncOffset={values.sync_offset} hideInfo setFieldValue={setFieldValue} setFieldTouched={setFieldTouched} />
             <Button primary type="submit" disabled={isSubmitting}>Load Dataset</Button>
             <Button onClick={onCancel} >Cancel</Button>
             <Button type="button" onClick={handleReset}>Reset</Button>
         </Form>
     )
 }
 
+const getInfoItemForDSPath = (ds_path: string, info?: DatasetInfoHDF5) => {
+    const dsItemsByPath: {
+        [k: string]: DatasetInfoHDF5Item
+    } = {};
+    info?.datasets?.forEach(dsItem => dsItemsByPath[dsItem.path] = dsItem);
+    // I think this will fail if datasets is undefined or empty
+    // yet the only possible value for ds_path comes from the dropdown
+    // menu which is itself defined by info.datasets so by definition
+    // it is valid!
```

Comment:
There's actually a case where this is not true: for some files, it can take a very long time to list the contained structure (groups, datasets, ...). Especially for matlab-generated HDF5 files, this can be the case, where often many small items are written to the file. To not block on this kind of file, we implemented a timeout and basically give up listing the datasets. Then, there is no drop-down and you have to manually enter the hdf5 dataset path.

Here is the current handling on the client side for this case:

https://github.com/LiberTEM/LiberTEM/blob/6042a5d3a28b6ec8177a392a5b736eaf76a3406e/client/src/dataset/components/HDF5ParamsForm.tsx#L64-L70

In that case, we can't really validate that the given shape(s) and offset is correct, as we don't have the info on them. As it is an edge case, I guess we can get away with handling errors / validation when the load button in pressed?

Reply:
OK, understood. I wish I could generate a test HDF5 file to see how that code path behaves! I guess I could force `isTimeOut`.

I've added some explicit handling of now that we actually use the dataset `info` after the creation of the dropdown/field. The changes now mean that the user's input in the Field is used to key into the info object which can result in `undefined` info items, which we then try to access for things like `nav_shape` (=> uncaught exception). I've checked explicitly for `undefined` but it might be better to refactor the handling of `Field`-type inputs.

Reply:
OK, looks good!